overlord/snapshotstate/backend/backend.go

```diff
@@ -178,6 +178,27 @@ func Filename(snapshot *client.Snapshot) string {
 	return filepath.Join(dirs.SnapshotsDir, fmt.Sprintf("%d_%s_%s_%s.zip", snapshot.SetID, snapshot.Snap, snapshot.Version, snapshot.Revision))
 }
 
+func setIDFromFilename(f string) (uint64, error) {
```

Comment:
I think it's unfortunate if the filename and the json disagree about the snapshot-id, We need to do something about this and this helper needs to go. I think @stolowski has some plans here fortunately :)

Reply:
Actually talking to @stolowski the filename will be the authority here and the json will contain "0" so this is good.